### PR TITLE
Fix EjectOnDeath pilot spawning.

### LINKED ISSUE #17286: Pilots spawned from aircraft killed above water freeze in the air
Regression from #17254.

### PR DESCRIPTION
**Pull request by pchote**
Fixes #17286 and cleans up some code quality issues.

Note that #17222 has not been merged to bleed yet, so the 5th-paratrooper-dies bug is still present and not related to this PR.

### CODE REVIEW

**Review by teinarss — APPROVED** [review]

File: OpenRA.Mods.Common/Traits/EjectOnDeath.cs [hunk 1]
```diff
@@ -50,58 +50,55 @@ void INotifyKilled.Killed(Actor self, AttackInfo e)
...
+				new LocationInit(self.Location),
+			};
+
+			// If airborne, offset the spawn location so the pilot doesn't drop on another infantry's head
+			var spawnPos = cp;
 			if (inAir)
 			{
-				self.World.AddFrameEndTask(w =>
+				var subCell = self.World.ActorMap.FreeSubCell(self.Location);
+				if (subCell != SubCell.Invalid)
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Traits/EjectOnDeath.cs]: Shouldn't we skip actor creation if `!Info.AllowUnsuitableCell` and `subCell == SubCell.Invalid`?

Comment (pchote) [reply OpenRA.Mods.Common/Traits/EjectOnDeath.cs]: We do, several lines above this.

Comment (abcdefg30) [reply OpenRA.Mods.Common/Traits/EjectOnDeath.cs]: I see, this is just to make sure we don't use `SubCell.Invalid` if we allow unsuitable cells.

**Comment by tovl** [comment]
How does this relate to #16353?

**Review by abcdefg30 — APPROVED** [review]

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/614d3a1c7b8113488a34070b689cbc8df9217f71...6192cc53a59e265c05a4d4bb7d951aa0d724642f)

https://github.com/OpenRA/OpenRA/commit/6c8b3e17de0322f3027a75100a09c963448a2acd

**Pull request merged**